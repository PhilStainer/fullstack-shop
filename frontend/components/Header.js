import React from 'react'
import Link from 'next/link'
import styled from 'styled-components'

import Topbar from '#root/components/Topbar'
import Search from '#root/components/Search'
import Actions from '#root/components/Actions'
import Cart from '#root/components/Cart'

const StyledMiddleBar = styled.div`
  display: grid;
  grid-template-columns: min-content 1fr auto min-content;
  align-items: center;
  grid-column-gap: 1rem;
  padding: 0 2rem;
  border-bottom: 1px solid ${props => props.theme.border};
`

const Logo = styled.h1`
  font-size: 3rem;
  position: relative;
  z-index: 2;
  a {
    padding: 0.5rem 1rem;
    text-transform: uppercase;
    text-decoration: none;
  }
`

const Header = () => {
  return (
    <>
      <Topbar />

      <StyledMiddleBar>
        <Logo>
          <Link href="/">
            <a>Shop</a>
          </Link>
        </Logo>

        <Search />

        <Actions />

        <Cart />
      </StyledMiddleBar>
    </>
  )
}

export default Header
