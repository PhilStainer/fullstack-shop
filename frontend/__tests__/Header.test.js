import {render} from '@testing-library/react'

import Header from '../components/Header'

test('matches snapshot', () => {
  const {asFragment} = render(<Header />)

  expect(asFragment()).toMatchSnapshot()
})
