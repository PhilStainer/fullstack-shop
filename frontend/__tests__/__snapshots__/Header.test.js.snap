// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <div
      class="Topbar__StyledTopbar-sc-15dzr1y-0 bDFGVI"
    >
      <ul>
        <li>
          <a
            href="/support"
          >
            Support
          </a>
        </li>
        <li>
          <a
            href="/contact"
          >
            Contact
          </a>
        </li>
        <li>
          <a
            href="/signin"
          >
            Sign in
          </a>
        </li>
      </ul>
    </div>
    <div
      class="Header__StyledMiddleBar-sc-1s2mubi-0 dFWmpV"
    >
      <h1
        class="Header__Logo-sc-1s2mubi-1 EVsmK"
      >
        <a
          href="/"
        >
          Shop
        </a>
      </h1>
      <div
        class="Search__StyledSearch-sexg82-0 pceGo"
      >
        <input
          placeholder="Search"
          type="text"
        />
      </div>
      <ul
        class="Actions__StyledActions-sc-1fjdd5g-0 jCVbsb"
      >
        <li>
          <a
            href="/sell"
          >
            Sell
          </a>
        </li>
        <li>
          <a
            href="/account"
          >
            Account
          </a>
        </li>
      </ul>
      <div>
        Â£2.00
      </div>
    </div>
  </div>
</DocumentFragment>
`;
